tests/backend/test_gaussian_gates.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""Unit tests for Gaussian gate"""
+
+import pytest
+import numpy as np
+from scipy.linalg import expm
+from scipy.stats import unitary_group
+from strawberryfields.backends.tfbackend.ops import choi_trick, n_mode_gate, single_mode_gate, two_mode_gate, gaussian_gate_matrix, gaussian_gate
+
+
+class TestUnitaryFunctionRelated:
+    """Basic tests over new functions related to gaussian gates"""
+    @pytest.mark.parametrize("num_mode", [4,5])
+    def test_choi_trick(self, num_mode, tol):
+        """Test if we can get correct C, mu, Sigma from S, d"""
+        W = unitary_group.rvs(num_mode)
+        V = unitary_group.rvs(num_mode)
+        r = np.random.random(num_mode) # r needs to be real
+        alpha = np.random.random(num_mode) + 1j * np.random.random(num_mode)
+        _C = np.exp(-0.5 * np.sum(np.abs(alpha) ** 2) - 0.5 * np.conj(alpha).T @ W @ np.diag(np.tanh(r)) @ W.T @ np.conj(alpha)) / np.sqrt(np.prod(np.cosh(r)))
+        _mu = np.block([ np.conj(alpha).T @ W @ np.diag(np.tanh(r)) @ W.T + alpha.T, -np.conj(alpha).T @ W @ np.diag(1/np.cosh(r)) @ V])
+        tanhr = np.diag(np.tanh(r))
+        sechr = np.diag(1 / np.cosh(r))
+        _Sigma = np.block(
+            [[W @ tanhr @ W.T, -W @ sechr @ V], [-V.T @ sechr @ W.T, -V.T @ tanhr @ V]])
+        OW = np.block([[W.real, -W.imag], [W.imag, W.real]])
+        OV = np.block([[V.real, -V.imag], [V.imag, V.real]])
+        S = OW@np.diag(np.concatenate([np.exp(-r), np.exp(r)]))@OV
+        expected_C, expected_mu, expected_Sigma = choi_trick(S, alpha, num_mode)
+        assert np.allclose(_C, expected_C, atol=tol, rtol=0)
+        assert np.allclose(_mu, expected_mu, atol=tol, rtol=0)
+        assert np.allclose(_Sigma, expected_Sigma, atol=tol, rtol=0)
+    
+    @pytest.mark.parametrize("cutoff", [5,7])
+    def test_n_mode_gate(self, cutoff, tol):
```

Comment:
I would not put more than one `assert` per test. It's good to have many, but maybe put them in different test functions?